Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift

```diff
@@ -117,6 +97,11 @@ open class ChatMessageListVC: _ViewController,
     open var isDateSeparatorEnabled: Bool {
         components.messageListDateSeparatorEnabled
     }
+
+    /// The message cell height caches. This makes sure that the message list doesn't
+    /// need to recalculate the cell height every time. This improve the scrolling
+    /// experience since the content size calculation is more precise.
+    private var cellHeightsCache: [MessageId: CGFloat] = [:]
```

Comment:
I'm not sure if the `MessageId` is the right key here. Imagine a case where we edit a message and add larger / smaller text. The id of the message would be the same, but the cell size not. 
Another thing to consider is if adding reactions / thread replies affects this size. 
To sum up, I would go with a computed `Differentiable` ID as a key instead.

Reply:
I think computing ids might be a bit more costly. Is there any way for us to clear the cache whenever the message is updated, somehow?

Reply:
It doesn't really matter when you edit the message or update the message. Keep in mind this is just the cell height estimation. It doesn't need to be exact, and not even close to the real height. This is only important when loading the previous messages. We can indeed check if a message was updated, and remove it from the cache, I actually did this already, but from my experience, this doesn't have any real improvement. So, for me, it does not make sense to make this more complex if we don't need to. For now, everything is working fine :)